scripts-available/CDB_TableUtils.sql

```diff
@@ -62,6 +61,13 @@ BEGIN
       );
     END IF;
 
+    -- Disable trigger for tracking updates to run manually
+    EXECUTE FORMAT(
+      'ALTER TABLE %I.%I DISABLE TRIGGER track_updates',
+      schema_name,
+      dest_table_name
+    );
+
     -- Truncate table
     EXECUTE FORMAT('TRUNCATE TABLE %I.%I', schema_name, dest_table_name);
 
```

Comment:
Note this is for a couple lines lower (line 75), but GitHub won't let me add this to the specific line.

This is not related to the deadlock.  However it'll prevent us from using this function in the importer.  The assumption for the temp table in the truncate/insert vs drop/replace logic is different.  In the truncate/insert logic below it provides the source_table_name without the schema (which therefore assumes it's in the public namespace).  However in the drop/replace logic it assumes the source_table_name is in the schema provided to the function (which can be different than public).  Can we please change the truncate/insert logic to match and provide the schema for the source table

```suggestion
EXECUTE FORMAT($q$
      INSERT INTO %I.%I ( %s )
        SELECT %s
        FROM %I.%I
    $q$, schema_name, dest_table_name, column_list, column_list, schema_name, source_table_name);
```

Reply:
Good observation - they probably should be consistent.  I will however have to verify if this change will work for all uses of this function in cartodb.

Note that not specifying a schema is not the same thing as specifying the `public` namespace.  Table resolution looks for the table in all schemas in the `search_path` configuration parameter, in order, and chooses the first. You may just need to set `search_path` to the correct value for your use case.

Reply:
Spoke with Tyler.  Although search_path will work he will change the truncate/insert to explicitly use the schema to match the drop/replace.  This will remove the perquisite for callers to set proper search_paths.